### ISSUE TITLE
Logging in with msn-pecan crashes pidgin

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dequis/msn-pecan#105]
```
What steps will reproduce the problem?
1. Log into msn with msn-pecan
2.
3.

What is the expected output? What do you see instead?
To log in. Pidgin crashes

Please use labels and text to provide additional information.

Debug log on enabling msn-pecan account:

(22:36:23) account: Connecting to account XXXXXXXXXXX@hotmail.com
(22:36:23) connection: Connecting. gc = 0x9e56368
(22:36:23) msn: new httpconn (0x9ccf918)
(22:36:23) g_log: pecan_contact_set_passport: assertion `contact' failed

...crash happens

gdb backtrace:

#0  0xb732229b in strlen () from /lib/tls/i686/cmov/libc.so.6
No symbol table info available.
#1  0xb75828ce in g_strdup () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#2  0xb5f0a890 in msn_session_set_password () from
/usr/local/lib/purple-2/libmsn-pecan.so
No locals.
#3  0xb5f03d60 in login () from /usr/local/lib/purple-2/libmsn-pecan.so
No locals.
#4  0xb7490cc8 in purple_connection_new (account=0x8745a50, regist=0,
password=0x87463a8 "PASSWORD_HIDDEN") at connection.c:170
        gc = (PurpleConnection *) 0x0
        prpl = <value optimized out>
        prpl_info = (PurplePluginProtocolInfo *) 0xb5f28980
        __PRETTY_FUNCTION__ = "purple_connection_new"
#5  0xb747afaa in purple_account_connect (account=0x8745a50) at account.c:1158
        prpl = <value optimized out>
        password = 0x13 <Address 0x13 out of bounds>
        __PRETTY_FUNCTION__ = "purple_account_connect"
#6  0xb74b42f5 in purple_prpl_change_account_status (account=0x8745a50,
old_status=0x8746308, new_status=0x8745e58) at prpl.c:324
        __PRETTY_FUNCTION__ = "purple_prpl_change_account_status"
#7  0xb74c3412 in purple_status_set_active_with_attrs_list
(status=0x8745e58, active=1, attrs=0x0) at status.c:662
        changed = 1
        l = <value optimized out>
        specified_attr_ids = (GList *) 0x0
        status_type = <value optimized out>
        __PRETTY_FUNCTION__ = "purple_status_set_active_with_attrs_list"
#8  0xb7479d6a in purple_account_set_status_list (account=0x8745a50,
status_id=0x8744dd0 "available", active=1, attrs=0x0)
    at account.c:1687
        status = (PurpleStatus *) 0x8745e58
        __PRETTY_FUNCTION__ = "purple_account_set_status_list"
#9  0xb7479e73 in purple_account_set_status (account=0x8745a50,
status_id=0x8744dd0 "available", active=1) at account.c:1662
        attrs = (GList *) 0x0
        id = <value optimized out>
        args = 0xbfa25310
"XS��\220\020�;�b��Q��\177�bPO�bXS��1�K���bPZt\b
\020\205K��S�����\220\020�\bv\b"
---Type <return> to continue, or q <return> to quit---
#10 0xb74b9717 in purple_savedstatus_activate_for_account
(saved_status=0x8e13be0, account=0x8745a50) at savedstatuses.c:1172
        status_type = (const PurpleStatusType *) 0x8745b28
        message = 0x0
        __PRETTY_FUNCTION__ = "purple_savedstatus_activate_for_account"
#11 0xb74ba131 in purple_savedstatus_activate (saved_status=0x8e13be0) at
savedstatuses.c:1126
        accounts = (GList *) 0x8e64f50
        node = (GList *) 0x8e67fb0
        old = (PurpleSavedStatus *) 0x87608c8
        __PRETTY_FUNCTION__ = "purple_savedstatus_activate"
#12 0x080ea6e1 in activate_currently_selected_status (status_box=0x8b7e008)
at gtkstatusbox.c:2484
        acct_status_type = (PurpleStatusType *) 0x0
        primitive = PURPLE_STATUS_AVAILABLE
        type = PIDGIN_STATUS_BOX_TYPE_PRIMITIVE
        data = (gpointer) 0x2
        title = <value optimized out>
        iter = {stamp = 1194539657, user_data = 0x88fefa8, user_data2 =
0x0, user_data3 = 0x0}
        path = <value optimized out>
        message = 0x0
        saved_status = (PurpleSavedStatus *) 0x8e13be0
#13 0x080ead59 in remove_typing_cb (status_box=0x8b7e008) at
gtkstatusbox.c:2585
No locals.
#14 0xb7561e26 in ?? () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#15 0xb75616f8 in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#16 0xb7564da3 in ?? () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#17 0xb75652c2 in g_main_loop_run () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#18 0xb78f83a9 in gtk_main () from /usr/lib/libgtk-x11-2.0.so.0
No symbol table info available.
#19 0x080c0ed8 in main (argc=Cannot access memory at address 0x13
) at gtkmain.c:888
        opt_help = <value optimized out>
 opt_login = 0
        opt_nologin = 0
        opt_version = <value optimized out>
        opt_si = 1
        opt_config_dir_arg = 0x0
        opt_login_arg = 0x0
        opt_session_arg = 0x0
        accounts = <value optimized out>
        sig_indx = <value optimized out>
        sigset = {__val = {82951, 0 <repeats 31 times>}}
        prev_sig_disp = <value optimized out>
        errmsg =
"�u���237�234u�����\237�003\000\000\000\002\000\000\000\20
4W����b\000\000\000\f\000\000\000��\b�����237�\b����
�\b�\f\000\000\000��\001\000\000\000\b���R�",
'\0' <repeats 20 times>,
"������&�234u��\000\000\000\000\035\000\000\000\210\000\000\00
0�u���u���237�005\000\000\000\004\000\000\000\000X����b\
000\000\000\024\000\000\000p�\b�����237�210�\b�p�\b�\024\0
00\000\000�237�234u�����\001\000\000\000�...
        segfault_message_tmp = <value optimized out>
        error = (GError *) 0x0
        opt = <value optimized out>
        gui_check = <value optimized out>
        debug_enabled = <value optimized out>
        migration_failed = <value optimized out>
        active_accounts = <value optimized out>
        long_options = {{name = 0x81072ca "config", has_arg = 1, flag =
0x0, val = 99}, {name = 0x80f6a7c "debug", has_arg = 0, 
    flag = 0x0, val = 100}, {name = 0x80fa0e4 "help", has_arg = 0, flag =
0x0, val = 104}, {name = 0x80f6d34 "login", has_arg = 2, 
    flag = 0x0, val = 108}, {name = 0x81036c8 "multiple", has_arg = 0, flag
= 0x0, val = 109}, {name = 0x81036d1 "nologin", 
    has_arg = 0, flag = 0x0, val = 110}, {name = 0x81036d9 "nocrash",
has_arg = 0, flag = 0x0, val = 120}, {
    name = 0x81072c0 "session", has_arg = 1, flag = 0x0, val = 115}, {name
= 0x80fbd4a "version", has_arg = 0, flag = 0x0, 
    val = 118}, {name = 0x81072d3 "display", has_arg = 1, flag = 0x0, val =
68}, {name = 0x81036e1 "sync", has_arg = 0, flag = 0x0, 
    val = 83}, {name = 0x0, has_arg = 0, flag = 0x0, val = 0}}



```

Original issue reported on code.google.com by `craigwharding` on 15 Dec 2008 at 2:49

**Comment by GoogleCodeExporter** [comment]
```
Does this happen all the time?
```

Original comment by `felipe.contreras` on 16 Dec 2008 at 7:44
- Added labels: **Type-Defect**, **Priority-Critical**, **Milestone-0.1.0**

**Comment by GoogleCodeExporter** [comment]
```
Yes, every time. I haven't used msn-pecan since 0.0.15 because of this crash. 
```

Original comment by `craigwharding` on 16 Dec 2008 at 10:30

**Comment by GoogleCodeExporter** [comment]
```
craigwharding: sorry about that.

It seems to be related to the independent pm option, maybe you can try changing 
that.
Also if you have valgrind running pidgin with it would be very helpful.
```

Original comment by `felipe.contreras` on 17 Dec 2008 at 12:40

**Comment by GoogleCodeExporter** [comment]
```
How can I change this PM option without logging in..? And what would I change 
it to?

Is it in the prefs.xml or accounts.xml file somewhere?
```

Original comment by `craigwharding` on 17 Dec 2008 at 2:14

**Comment by GoogleCodeExporter** [comment]
```
pidgin --nologin

Or edit accounts.xml and set use_independent_pm to 0.
```

Original comment by `felipe.contreras` on 17 Dec 2008 at 9:46

**Comment by GoogleCodeExporter** [comment]
```
Craig, can you try with http://github.com/dfiloni/msn-pecan/tree/master if the 
bug is
still there?
```

Original comment by `devid...@gmail.com` on 17 Dec 2008 at 4:40

**Comment by GoogleCodeExporter** [comment]
```
Is there a revision I can work with that will compile? The current one won't 
compile
for me.

session.c: In function ‘msn_session_finish_login’:
session.c:477: warning: assignment discards qualifiers from pointer target type
[CC] switchboard.o
switchboard.c: In function ‘got_voice_clip’:
switchboard.c:1249: error: ‘decoded_file’ undeclared (first use in this 
function)
switchboard.c:1249: error: (Each undeclared identifier is reported only once
switchboard.c:1249: error: for each function it appears in.)
switchboard.c:1249: warning: too many arguments for format
switchboard.c:1234: warning: ignoring return value of ‘fwrite’, declared 
with
attribute warn_unused_result

```

Original comment by `craigwharding` on 19 Dec 2008 at 3:10

**Comment by GoogleCodeExporter** [comment]
```
I just fixed it, sorry for the error.
```

Original comment by `devid...@gmail.com` on 19 Dec 2008 at 2:32

**Comment by GoogleCodeExporter** [comment]
```
Using an updated revision, it compiles but pidgin still crashes... backtrace 
below
(Too bad files cannot be attached)

#0  purple_proxy_info_get_type (info=0x72616867) at proxy.c:166
        __PRETTY_FUNCTION__ = "purple_proxy_info_get_type"
#1  0xb7512567 in purple_proxy_get_setup (account=0xa821cb0) at proxy.c:2043
        gpi = (PurpleProxyInfo *) 0x72616867
        tmp = <value optimized out>
        tmp_none_proxy_info = (PurpleProxyInfo *) 0xa81c0e0
#2  0xb7512d94 in purple_proxy_connect (handle=0x0, account=0xa821cb0, 
    host=0xa129ec8 "messenger.hotmail.com", port=80, 
    connect_cb=0xb69e5650 <connect_cb>, data=0xa81c9b8) at proxy.c:2130
        connecthost = <value optimized out>
        connectport = <value optimized out>
        connect_data = (PurpleProxyConnectData *) 0xa7bd638
        __PRETTY_FUNCTION__ = "purple_proxy_connect"
#3  0xb69e4dc0 in msn_httpconn_connect (httpconn=0xa81c9b8, 
    host=0xa129ec8 "messenger.hotmail.com", port=1863) at httpconn.c:748
        __PRETTY_FUNCTION__ = "msn_httpconn_connect"
#4  0xb69f5eef in msn_servconn_connect (servconn=0xa81c008, 
    host=0xa129ec8 "messenger.hotmail.com", port=1863, force=1)
    at servconn.c:227
        session = <value optimized out>
        __PRETTY_FUNCTION__ = "msn_servconn_connect"
#5  0xb69f1b74 in msn_notification_connect (notification=0xa272678, 
    host=0xa129ec8 "messenger.hotmail.com", port=1863) at notification.c:135
---Type <return> to continue, or q <return> to quit---
        servconn = (MsnServConn *) 0xa81c008
        __PRETTY_FUNCTION__ = "msn_notification_connect"
#6  0xb69f7262 in msn_session_connect (session=0xa81c940, 
    host=0xa129ec8 "messenger.hotmail.com", port=1863, http_method=-1219657425)
    at session.c:121
        __PRETTY_FUNCTION__ = "msn_session_connect"
#7  0xb5f55dc6 in login () from /usr/local/lib/purple-2/libmsn-pecan.so
No locals.
#8  0xb74f1cc8 in purple_connection_new (account=0xa128c70, regist=0, 
    password=0xa129d00 "$PASSWORD") at connection.c:170
        gc = (PurpleConnection *) 0xa81c940
        prpl = <value optimized out>
        prpl_info = (PurplePluginProtocolInfo *) 0xb5f80a20
        __PRETTY_FUNCTION__ = "purple_connection_new"
#9  0xb74dbfaa in purple_account_connect (account=0xa128c70) at account.c:1158
        prpl = <value optimized out>
        password = 0x72616867 <Address 0x72616867 out of bounds>
        __PRETTY_FUNCTION__ = "purple_account_connect"
#10 0xb75152f5 in purple_prpl_change_account_status (account=0xa128c70, 
    old_status=0xa129c50, new_status=0xa129640) at prpl.c:324
        __PRETTY_FUNCTION__ = "purple_prpl_change_account_status"
#11 0xb7524412 in purple_status_set_active_with_attrs_list (status=0xa129640, 
    active=1, attrs=0xa499c00) at status.c:66
 changed = 1
        l = <value optimized out>
        specified_attr_ids = (GList *) 0xa49a530
        status_type = <value optimized out>
        __PRETTY_FUNCTION__ = "purple_status_set_active_with_attrs_list"
#12 0xb74dad6a in purple_account_set_status_list (account=0xa128c70, 
    status_id=0xa128e18 "away", active=1, attrs=0xa499c00) at account.c:1687
        status = (PurpleStatus *) 0xa129640
        __PRETTY_FUNCTION__ = "purple_account_set_status_list"
#13 0xb74dae73 in purple_account_set_status (account=0xa128c70, 
    status_id=0xa128e18 "away", active=1) at account.c:1662
        attrs = (GList *) 0xa499c00
        id = <value optimized out>
        args = 0xbfd88e68 "����\017X� {|\n��\024\n�\216ؿ1�Q��\021\np\214\022\n\0\225Q�"

#14 0xb751a6e2 in purple_savedstatus_activate_for_account (
    saved_status=0xa1145d8, account=0xa128c70) at savedstatuses.c:1167
        status_type = (const PurpleStatusType *) 0xa128df0
        message = 0xa133898 "pm"
        __PRETTY_FUNCTION__ = "purple_savedstatus_activate_for_account"

#15 0xb751b131 in purple_savedstatus_activate (saved_status=0xa1145d8)
    at savedstatuses.c:1126
        accounts = (GList *) 0xa14beb0
---Type <return> to continue, or q <return> to quit---
        node = (GList *) 0xa7c7b20
        old = (PurpleSavedStatus *) 0xa14bd58
        __PRETTY_FUNCTION__ = "purple_savedstatus_activate"


#16 0x080ea6e1 in activate_currently_selected_status (status_box=0xa562000)
    at gtkstatusbox.c:2484
        acct_status_type = (PurpleStatusType *) 0x0
        primitive = PURPLE_STATUS_AWAY
        type = PIDGIN_STATUS_BOX_TYPE_PRIMITIVE
        data = (gpointer) 0x5
        title = <value optimized out>
        iter = {stamp = 688162724, user_data = 0xa2e5790, user_data2 = 0x0, 
  user_data3 = 0x0}
        path = <value optimized out>
        message = 0xa671bb8 "pm"
        saved_status = (PurpleSavedStatus *) 0xa1145d8
#17 0x080ead59 in remove_typing_cb (status_box=0xa562000)
    at gtkstatusbox.c:2585
No locals.
#18 0xb75c2e26 in ?? () from /usr/lib/libglib-2.0.so.0
No symbol table info available.

#19 0xb75c26f8 in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#20 0xb75c5da3 in ?? () from /usr/lib/libglib-2.0.so.0
---Type <return> to continue, or q <return> to quit---
No symbol table info available.
#21 0xb75c62c2 in g_main_loop_run () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#22 0xb79593a9 in gtk_main () from /usr/lib/libgtk-x11-2.0.so.0
No symbol table info available.
#23 0x080c0ed8 in main (argc=Cannot access memory at address 0x0
) at gtkmain.c:888
        opt_help = <value optimized out>
        opt_login = 0
        opt_nologin = 0
        opt_version = <value optimized out>
        opt_si = 1
        opt_config_dir_arg = 0x0
        opt_login_arg = 0x0
        opt_session_arg = 0x0
        accounts = <value optimized out>
        sig_indx = <value optimized out>
        sigset = {__val = {82951, 0 <repeats 31 times>}}
        prev_sig_disp = <value optimized out>
        errmsg = "\017�ؿ���������003\000\000\000\002\000\000\000�222ؿ��b\000\000 
\000\000\000��016�\034�ؿ˯�\016���016�\f\000\000\000��\00
1\000\000\000\b���212X    0'
<repeats 20 times>, 
"���\223ؿ�6��ؿ\000\000\000\000\035\000\000\000\210\00     
017�ؿ\b�ؿ��\005\000\000\000\004\000\000\000P\223ؿ��b\000\000\000\
024\000\0p�016�\034�ؿ˯�210�0---Type
<return> to continue, or q <return> to quit---
16�p�016�\024\000\000\000����\034�ؿ\001\000\000\000�"...
        segfault_message_tmp = <value optimized out>
        error = (GError *) 0x0
        opt = <value optimized out>
        gui_check = <value optimized out>
        debug_enabled = <value optimized out>
        migration_failed = <value optimized out>
        active_accounts = <value optimized out>
        long_options = {{name = 0x81072ca "config", has_arg = 1, flag = 0x0, 
    val = 99}, {name = 0x80f6a7c "debug", has_arg = 0, flag = 0x0, val = 100}, 
  {name = 0x80fa0e4 "help", has_arg = 0, flag = 0x0, val = 104}, {
    name = 0x80f6d34 "login", has_arg = 2, flag = 0x0, val = 108}, {
    name = 0x81036c8 "multiple", has_arg = 0, flag = 0x0, val = 109}, {
    name = 0x81036d1 "nologin", has_arg = 0, flag = 0x0, val = 110}, {
    name = 0x81036d9 "nocrash", has_arg = 0, flag = 0x0, val = 120}, {
    name = 0x81072c0 "session", has_arg = 1, flag = 0x0, val = 115}, {
    name = 0x80fbd4a "version", has_arg = 0, flag = 0x0, val = 118}, {
    name = 0x81072d3 "display", has_arg = 1, flag = 0x0, val = 68}, {
    name = 0x81036e1 "sync", has_arg = 0, flag = 0x0, val = 83}, {name = 0x0, 
    has_arg = 0, flag = 0x0, val = 0}}


```

Original comment by `craigwharding` on 19 Dec 2008 at 9:09

**Comment by GoogleCodeExporter** [comment]
```
[QUOTE]Yes, every time. I haven't used msn-pecan since 0.0.15 because of this 
crash.
[/QUOTE]

So the crash is not related to independent pm option.
```

Original comment by `devid...@gmail.com` on 20 Dec 2008 at 2:17

**Comment by GoogleCodeExporter** [comment]
```
I can add some debug print statements and then repost.. If you want me to add 
some
purple_debug_info char statements, let me know what variables you want to see..
```

Original comment by `craigwharding` on 20 Dec 2008 at 2:39

**Comment by GoogleCodeExporter** [comment]
```
devidfil: why do you say it's not related to pm?

craig meant every time he tries to log in, afaik he hasn't reported what 
happens with
the option on, and with the option off.
```

Original comment by `felipe.contreras` on 29 Dec 2008 at 8:13

**Comment by GoogleCodeExporter** [comment]
```
felipe: the new option for pm is implemented from 0.0.17 release but craig said 
that
this issue is related to 0.0.16+
```

Original comment by `devid...@gmail.com` on 29 Dec 2008 at 8:18

**Comment by GoogleCodeExporter** [comment]
```
Ahh, sorry, I didn't notice that.

craig: It would be great if you can bisect the issue, basically do this:

git bisect start 0.0.17 0.0.15

(compile, install, test)

if it works:
git bisect good

if it fails:
git bisect bad

(compile, install, test)

And repeat until you find the problematic commit.
```

Original comment by `felipe.contreras` on 29 Dec 2008 at 9:03

**Comment by GoogleCodeExporter** [comment]
```
This has worked on the first bisect. I can log into my acct using msn-pecan. 
See good
bisect output below. Thanks.

$ git bisect good
Bisecting: 17 revisions left to test after this
[8eb8691e7c49be776508903b9cf9138ac5c2895d] Add new option for custom personal me
ssage.

```

Original comment by `craigwharding` on 5 Jan 2009 at 4:10

**Comment by GoogleCodeExporter** [comment]
```
Craig: can you test http://github.com/dfiloni/msn-pecan/tree/master ? That 
option is
disabled by default ("Use personal status messages" option is now TRUE) so if 
it is
the problem you should be able to connect now.
```

Original comment by `devid...@gmail.com` on 5 Jan 2009 at 10:03

**Comment by GoogleCodeExporter** [comment]
```
craig: the idea is to find the bad commit in order to fix it in the latest 
revision.

Can you keep doing bisects until you find the bad commit?
```

Original comment by `felipe.contreras` on 5 Jan 2009 at 10:12

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
For some reason I get a different backtrace log after doing the very first 
compile
and install from 'git bisect start 0.0.17 0.0.15'

See attached backtrace log.
```

Original comment by `craigwharding` on 12 Jan 2009 at 8:23

Attachments:
- [msn_pecan_backtrace.log](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-105/comment-19/msn_pecan_backtrace.log)

**Comment by GoogleCodeExporter** [comment]
```
Doing git bisect bad, I get this.

$ git bisect bad
ac09660e6eac040025aee7f167afb474fc38ecbc is first bad commit
commit ac09660e6eac040025aee7f167afb474fc38ecbc
Author: Felipe Contreras <felipe.contreras@gmail.com>
Date:   Fri Aug 29 12:54:00 2008 +0300

    Fix win32 build.

:040000 040000 387402016b153cc446ba4167a951f6e4ee6d3a50
f8983aafae3f30b191a744aacb66e8c8cf18f14e M      cmd
:100644 100644 a42034135f8c6d749509e83746abbe4ea993ed25
a755e200e04647d9cebe8eaabbb4e55fd2c1dfe5 M      fix_purple.h
:040000 040000 f5dc896144baed1c33c87ec5d4e9f8a9d4bb0986
f83a87b5008ac8b0e816fb21a3c47c07c985d4c2 M      io
:100644 100644 1f91b0bdd9411981ba412287eca87f661eb49587
6df64d1acbecad81a478c4caaaeba431c090837d M      pecan_oim.c

```

Original comment by `craigwharding` on 12 Jan 2009 at 8:53

**Comment by GoogleCodeExporter** [comment]
```
I too can not get this plugin to work. 
Running 32 bit Intrepid and connecting to MSN via http via a squid proxy.
I've tried .17, .15 and .10 and they all have the problem.

Removing ~/.purple and trying again makes no difference, so I can't actually 
find a
working version to find when it broke.

Problem appears with 2.5.2-0ubuntu1 and also 2.5.4 compiled from source.

here is a backtrace with .17 against 2.5.2-0ubuntu1.
Clean .purple directory.

Starting program: /usr/bin/pidgin                                            
[Thread debugging using libthread_db enabled]                                
[New Thread 0xb7026700 (LWP 21729)]                                          

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0xb7026700 (LWP 21729)]        
0xb605028f in read_impl () from /usr/local/lib/purple-2/libmsn-pecan.so
Current language:  auto; currently asm                                 
(gdb) bt full                                                          
#0  0xb605028f in read_impl () from /usr/local/lib/purple-2/libmsn-pecan.so
No locals.                                                                 
#1  0xb604d9bd in pecan_node_read () from 
/usr/local/lib/purple-2/libmsn-pecan.so
No locals.                                                                      

#2  0xb6051289 in read_cb () from /usr/local/lib/purple-2/libmsn-pecan.so       

No locals.                                                                      

#3  0xb77ac6fd in ?? () from /usr/lib/libglib-2.0.so.0                          

No symbol table info available.                                                 

#4  0xb77756f8 in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0     

No symbol table info available.                                                 

#5  0xb7778da3 in ?? () from /usr/lib/libglib-2.0.so.0                          

No symbol table info available.                                                 

#6  0xb77792c2 in g_main_loop_run () from /usr/lib/libglib-2.0.so.0             

No symbol table info available.                                                 

#7  0xb7a843a9 in gtk_main () from /usr/lib/libgtk-x11-2.0.so.0                 

No symbol table info available.                                                 

#8  0x080caaaf in main (argc=1, argv=0xbf943624) at ../../pidgin/gtkmain.c:887  

        opt_help = 0                                                             
        opt_login = 0                                                            
        opt_nologin = 0                                                          
        opt_nocrash = 0                                                          
        opt_version = 0                                                          
        opt_si = 1                                                               
        opt_config_dir_arg = 0x0                                                 
---Type <return> to continue, or q <return> to quit---                          

        opt_login_arg = 0x0                                                      
        opt_session_arg = 0x0                                                    
        search_path = 0x8461c40 "\001"                                           
        accounts = (GList *) 0x14                                                
        sig_indx = 1                                                             
        sigset = {__val = {81927, 0 <repeats 31 times>}}
        prev_sig_disp = (void (*)(int)) 0
        errmsg = "�{\034��\222����e�\000\000\000\000\232\223d�
`���3\224��3\224��3\224������\004���C\005�\000
\000\000\000\016\000\000\000�_���3\224��_��\004\000\000\000\003\
000\000\000�\025\224�����\b\000\000\000\020\000\000\000x\003\005��
��3\224��_��\214\003\005�x\003\005�\020",
'\0' <repeats 11 times>, "�\222��8�9���e�", '\0' <repeats 12 
times>,
"�3\002\000�3\224��3\224��3\224�����,8\002\000,H\002\000\0
00\000\000\000�\027\000\000�_���3\224��_��\004\000\000\000\003
\000\000"...
        segfault_message_tmp = 0x84413c8 ""
        error = (GError *) 0x0
        opt = -1
        gui_check = 1
        debug_enabled = 0
        migration_failed = 0
        active_accounts = (GList *) 0x0
        long_options = {{name = 0x8114a5b "config", has_arg = 1, flag = 0x0, val = 99}, {
    name = 0x8114a62 "debug", has_arg = 0, flag = 0x0, val = 100}, {name = 0x8114a68
"help",
    has_arg = 0, flag = 0x0, val = 104}, {name = 0x8114a6d "login", has_arg = 2, flag
= 0x0,
    val = 108}, {name = 0x8114a73 "multiple", has_arg = 0, flag = 0x0, val = 109}, {
    name = 0x8114a7c "nologin", has_arg = 0, flag = 0x0, val = 110}, {name =
0x8114a84 "nocrash",
---Type <return> to continue, or q <return> to quit---
    has_arg = 0, flag = 0x0, val = 120}, {name = 0x8114a8c "session", has_arg = 1,
flag = 0x0,
    val = 115}, {name = 0x811478a "version", has_arg = 0, flag = 0x0, val = 118}, {
    name = 0x8114a94 "display", has_arg = 1, flag = 0x0, val = 68}, {name = 0x8114a9c
"sync",
    has_arg = 0, flag = 0x0, val = 83}, {name = 0x0, has_arg = 0, flag = 0x0, val = 0}}


The output from pidgin -d is:
(13:14:06) proxy: Connection in progress
(13:14:06) proxy: Connected to gateway.messenger.hotmail.com:80.
(13:14:06) proxy: HTTP proxy connection established
(13:14:06) msn: cmd/cmdproc.c:98:show_debug_cmd() C: 000: VER 1 MSNP12 CVR0
(13:14:06) msn: io/pecan_http_server.c:650:read_impl() adding child: 0x9d6f8f0
(13:14:06) msn: cmd/cmdproc.c:98:show_debug_cmd() S: 000: VER 1 MSNP12
(13:14:06) msn: cmd/cmdproc.c:98:show_debug_cmd() C: 000: CVR 2 0x0409 winnt 
5.1 i386
MSNMSGR 6.0.0602MSMSGS matt@matt-foster.me.uk
(13:14:06) GLib: g_strsplit: assertion `string != NULL' failed
dns[22102]: Oops, father has gone, wait for me, wait...!
Segmentation fault (core dumped)


I wonder if this is something to do with the glib version that comes on 
Intrepid?
libglib2.0-0                               2.18.2-0ubuntu2

Thanks,

Matt

```

Original comment by `Matt.Fos...@gmail.com` on 21 Jan 2009 at 1:17

**Comment by GoogleCodeExporter** [comment]
```
matt: your issue seems to be different.

But just to make sure, craig: are you using http method?
```

Original comment by `felipe.contreras` on 21 Jan 2009 at 6:44

**Comment by GoogleCodeExporter** [comment]
```
I do not have that ticked.
```

Original comment by `craigwharding` on 21 Jan 2009 at 6:59

**Comment by GoogleCodeExporter** [comment]
```
matt: please fill another issue report, this is different.
```

Original comment by `felipe.contreras` on 21 Jan 2009 at 7:05

**Comment by GoogleCodeExporter** [comment]
```
I can log in using the 0.18, marking as fixed.
```

Original comment by `craigwharding` on 16 Feb 2009 at 8:34
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
huh? strange! I don't recall any change that could have fixed that =/
```

Original comment by `felipe.contreras` on 16 Feb 2009 at 8:49
